### PR TITLE
fix(nodebuilder/header): Cancel subscription when done

### PR DESCRIPTION
**Pull request by renaynay**
Found while debugging swamp.

We never cancel subscription so when node attempts to shut down, there are outstanding subs that need to be closed.

### CODE REVIEW

**Review by Wondertan — APPROVED** [review]

**Comment by Wondertan** [comment]
🤠

**Review by vgonkivs — DISMISSED** [review]
🤠

**Review by vgonkivs — APPROVED** [review]

**Review by Wondertan — CHANGES_REQUESTED** [review]
Re file name. That's API test, not an rpc tesy

File: nodebuilder/tests/rpc_test.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	// subscribe to headers via the light node's RPC header subscription
+	subctx, subcancel := context.WithCancel(ctx)
+	sub, err := light.HeaderServ.Subscribe(subctx)
+	require.NoError(t, err)
+	// listen for 5 headers
+	for i := 0; i < 5; i++ {
+		if ctx.Err() != nil {
+			t.Fatal(ctx.Err())
+		}
+		<-sub
```
[the remark this answers is not in the document]
Comment (vgonkivs) [reply nodebuilder/tests/rpc_test.go:48]: 🚔

[the remark this answers is not in the document]
Comment (distractedm1nd) [reply nodebuilder/tests/rpc_test.go:48]: Wait why? Interesting, didnt know this

**Review by distractedm1nd — APPROVED** [review]

File: nodebuilder/tests/rpc_test.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Wondertan) [reply nodebuilder/tests/rpc_test.go:48]: It may if sub is not giving us anything

[the remark this answers is not in the document]
Comment (renaynay) [reply nodebuilder/tests/rpc_test.go:48]: very tru

**Review by Wondertan — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]
yoloACK

**Pull request merged**